core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -268,11 +283,22 @@ import com.datastax.driver.core.utils.Bytes
         }
         val pnr = if (switchPartition) partition + 1 else partition
         queryState = QueryInProgress(switchPartition, fetchMore = false, System.nanoTime())
-        log.debug(
-          "EventsByPersistenceId [{}] Query from seqNr [{}] in partition [{}]",
-          persistenceId, seqNr, pnr
-        )
-        session.selectEventsByPersistenceId(persistenceId, pnr, seqNr, toSeqNr, fetchSize)
+
+        val endNr = lookingForMissingSeqNr match {
+          case Some(_) =>
+            log.debug(
+              "EventsByPersistenceId [{}] Query for missing seqNr [{}] in partition [{}]",
+              persistenceId, seqNr, pnr
+            )
+            seqNr
+          case _ =>
+            log.debug(
+              "EventsByPersistenceId [{}] Query from seqNr [{}] in partition [{}]",
+              persistenceId, seqNr, pnr
+            )
+            toSeqNr
```

Comment:
it's only the logging that is different so you can remove `endNr`

Reply:
No, it's using `toSeqNr` in the "normal" case and `seqNr` as end in the look-for-exactly-missing-mode.

Btw. Had a discussion with Christopher about if we need a separate query or if using a span with same start as end will do it, it is likely not optimized by C* but as this should happen seldom it may not be a problem.

Reply:
ah, misread